drivers/rtc/rtc_stm32.c

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (c) 2018 Workaround GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Source file for the STM32 RTC driver
+ *
+ */
+
+#include <errno.h>
+
+#include <device.h>
+#include <init.h>
+#include <kernel.h>
+#include <rtc.h>
+#include <power.h>
+#include <soc.h>
+#include <interrupt_controller/exti_stm32.h>
+
+#ifdef CONFIG_SOC_SERIES_STM32F4X
+#define IRQ_RTC_ALARM STM32F4_IRQ_RTC_ALARM
+#endif
+#ifdef CONFIG_SOC_SERIES_STM32L4X
+#define IRQ_RTC_ALARM STM32L4_IRQ_RTC_ALARM
+#endif
+
+/* Configuration data */
+struct rtc_stm32_config {
+	u32_t rtc_base;
+};
+
+/* Runtime driver data */
+struct rtc_stm32_data {
+	/* RTC peripheral handler */
+	RTC_HandleTypeDef hrtc;
+	void (*cb_fn)(struct device *dev);
+#ifdef CONFIG_RTC_STM32_API_REENTRANCY
+	struct k_sem sem;
+#endif
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+	u32_t device_power_state;
+#endif
+};
+
+
+static struct rtc_stm32_data rtc_data;
+
+#ifdef CONFIG_RTC_STM32_API_REENTRANCY
+#define RP_GET(dev) (&((struct rtc_stm32_data *)(dev->driver_data))->sem)
+#else
+#define RP_GET(dev) (NULL)
+#endif
+
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+
+static void rtc_stm32_set_power_state(struct device *dev, u32_t power_state)
+{
+	struct rtc_stm32_data *context = dev->driver_data;
+
+	context->device_power_state = power_state;
+}
+
+static u32_t rtc_stm32_get_power_state(struct device *dev)
+{
+	struct rtc_stm32_data *context = dev->driver_data;
+
+	return context->device_power_state;
+}
+#else
+#define rtc_stm32_set_power_state(...)
+#endif
+static int rtc_stm32_set_alarm(struct device *dev, const u32_t alarm_val);
+
+void HAL_RTC_AlarmAEventCallback(RTC_HandleTypeDef *hrtc)
+{
+}
+
+static void rtc_stm32_enable(struct device *dev)
+{
+	__HAL_RCC_RTC_ENABLE();
+}
+
+static void rtc_stm32_disable(struct device *dev)
+{
+	__HAL_RCC_RTC_DISABLE();
+}
+
+
+static int rtc_stm32_set_config(struct device *dev, struct rtc_config *cfg)
+{
+	struct rtc_stm32_data *context = dev->driver_data;
+	RTC_TimeTypeDef sTime;
+	RTC_DateTypeDef sDate;
+	int result = 0;
+
+	if (cfg->cb_fn != NULL) {
+		context->cb_fn = cfg->cb_fn;
+	}
+	sDate.Year = 0;
+	sDate.Month = 0;
+	sDate.Date = 0;
+	sDate.WeekDay = 0;
+
+	if (IS_ENABLED(CONFIG_RTC_STM32_API_REENTRANCY)) {
+		k_sem_take(RP_GET(dev), K_FOREVER);
+	}
+
+	result = HAL_RTC_SetDate(&context->hrtc, &sDate, RTC_FORMAT_BIN);
+	if (result) {
+		result = -EIO;
+	}
+	sTime.Hours      = ((cfg->init_val/1000)/3600)%24;
```

Comment:
shouldn't you add space between operators (checkpatch might complain afaik)

Reply:
Done (I was also surprised checkpatch did not complain on my machine)